rticonnextdds_connector.go

```diff
@@ -444,139 +460,151 @@ func (input *Input) Take() error {
+	var retVal C.double
+	err := samples.getNumber(index, fieldName, &retVal)
+	return uint8(retVal), err
 }
 
 // GetUint16 is a function to retrieve a value of type uint16 from the samples
-func (samples *Samples) GetUint16(index int, fieldName string) uint16 {
-	fieldNameCStr := C.CString(fieldName)
-	defer C.free(unsafe.Pointer(fieldNameCStr))
-	return uint16(C.RTIDDSConnector_getNumberFromSamples(unsafe.Pointer(samples.input.connector.native), samples.input.nameCStr, C.int(index+1), fieldNameCStr))
+func (samples *Samples) GetUint16(index int, fieldName string) (uint16, error) {
+	var retVal C.double
+	err := samples.getNumber(index, fieldName, &retVal)
+	return uint16(retVal), err
 }
 
 // GetUint32 is a function to retrieve a value of type uint32 from the samples
-func (samples *Samples) GetUint32(index int, fieldName string) uint32 {
-	fieldNameCStr := C.CString(fieldName)
-	defer C.free(unsafe.Pointer(fieldNameCStr))
-	return uint32(C.RTIDDSConnector_getNumberFromSamples(unsafe.Pointer(samples.input.connector.native), samples.input.nameCStr, C.int(index+1), fieldNameCStr))
+func (samples *Samples) GetUint32(index int, fieldName string) (uint32, error) {
+	var retVal C.double
+	err := samples.getNumber(index, fieldName, &retVal)
+	return uint32(retVal), err
 }
 
 // GetUint64 is a function to retrieve a value of type uint64 from the samples
-func (samples *Samples) GetUint64(index int, fieldName string) uint64 {
-	fieldNameCStr := C.CString(fieldName)
-	defer C.free(unsafe.Pointer(fieldNameCStr))
-	return uint64(C.RTIDDSConnector_getNumberFromSamples(unsafe.Pointer(samples.input.connector.native), samples.input.nameCStr, C.int(index+1), fieldNameCStr))
+func (samples *Samples) GetUint64(index int, fieldName string) (uint64, error) {
+	var retVal C.double
+	err := samples.getNumber(index, fieldName, &retVal)
+	return uint64(retVal), err
 }
 
 // GetInt8 is a function to retrieve a value of type int8 from the samples
-func (samples *Samples) GetInt8(index int, fieldName string) int8 {
-	fieldNameCStr := C.CString(fieldName)
-	defer C.free(unsafe.Pointer(fieldNameCStr))
-	return int8(C.RTIDDSConnector_getNumberFromSamples(unsafe.Pointer(samples.input.connector.native), samples.input.nameCStr, C.int(index+1), fieldNameCStr))
+func (samples *Samples) GetInt8(index int, fieldName string) (int8, error) {
+	var retVal C.double
+	err := samples.getNumber(index, fieldName, &retVal)
+	return int8(retVal), err
 }
 
 // GetInt16 is a function to retrieve a value of type int16 from the samples
-func (samples *Samples) GetInt16(index int, fieldName string) int16 {
-	fieldNameCStr := C.CString(fieldName)
-	defer C.free(unsafe.Pointer(fieldNameCStr))
-	return int16(C.RTIDDSConnector_getNumberFromSamples(unsafe.Pointer(samples.input.connector.native), samples.input.nameCStr, C.int(index+1), fieldNameCStr))
+func (samples *Samples) GetInt16(index int, fieldName string) (value int16, err error) {
```

Comment:
So for those that you do.. `(value int16, err error)` should be changed to `(int16, error)` instead. This will make things a lot easier. And also since you don't use the `value` variable, make no sense to have it that way.

Reply:
Updated. 

https://github.com/rticommunity/rticonnextdds-connector-go/pull/37/commits/d558ef18ebf04e2ee82b6d8579b4fe685d36dbed

Reply:
Nice Thanks!